primitives/io/src/lib.rs

```diff
@@ -305,7 +306,7 @@ pub trait DefaultChildStorage {
 	/// Use this function to distribute the deletion of a single child trie across multiple
 	/// blocks.
 	#[version(2)]
-	fn storage_kill(&mut self, storage_key: &[u8], limit: Option<u32>) -> bool {
+	fn storage_kill(&mut self, storage_key: &[u8], limit: Option<u32>) -> (bool, u32) {
 		let child_info = ChildInfo::new_default(storage_key);
 		self.kill_child_storage(&child_info, limit)
 	}
```

Comment:
Two comments:

1. We are not allowed to change an existing runtime interface. We need to create a `#[version(3)]` with the new signature and leave version 2 intact.
2. One thing that came up during version2 review but came to late is that we should move the nicer type of the frame wrapper to the runtime interface. So it would be nice to already use the enum in the version3 of the runtime interface.

Reply:
I would need to implement `PassBy` to support this function returning `KillOutcome`. Do we still want that? seems pretty specific

Reply:
Yeah we want this ;) This would have been my comment here as well. We can move this type `KillOutcome` to somewhere else, outside of `frame-support`.

There are also derive macros for `PassBy` derive.